### PR TITLE
Improvement/modular renderer

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Markdown renderer

### Description
This PR splits the markdown renderer into a base and an extended renderer

### Steps

- [x] added implementation
- [ ] added / updated tests
- [ ] added / updated documentation
- [ ] extended changelog

### CODE REVIEW

File: src/components/editor/app-bar/help-button/help-button.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { Cheatsheet } from './cheatsheet'
 import { Links } from './links'
 import { Shortcut } from './shortcuts'
 
-enum HelpTabStatus {
+export enum HelpTabStatus {
```
Comment (ErikMichelson) [inline src/components/editor/app-bar/help-button/help-button.tsx:9]: Why does this need to be exported?

File: src/components/markdown-renderer/basic-markdown-renderer.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import './markdown-renderer.scss'
+import { combinedEmojiData } from './markdown-it-plugins/emoji/mapping'
+import { linkifyExtra } from './markdown-it-plugins/linkify-extra'
+import { MarkdownItParserDebugger } from './markdown-it-plugins/parser-debugger'
+import { ComponentReplacer } from './replace-components/ComponentReplacer'
+import { buildTransformer } from './utils/html-react-transformer'
+import { calculateNewLineNumberMapping } from './utils/line-number-mapping'
+import { LineKeys } from './utils/types'
+
+export interface AdditionalMarkdownRendererProps {
```
Comment (ErikMichelson) [inline src/components/markdown-renderer/basic-markdown-renderer.tsx:27]: Maybe this should be moved into a separate type definition file?

File: src/components/editor/app-bar/help-button/help-button.tsx (hunk 1 above)
Comment (mrdrogdrog) [reply src/components/editor/app-bar/help-button/help-button.tsx:9]: If I don't do that, then the linter shows a warning, that HelpTabStatus is unused.

File: src/components/markdown-renderer/basic-markdown-renderer.tsx (hunk 2 above)
Comment (mrdrogdrog) [reply src/components/markdown-renderer/basic-markdown-renderer.tsx]: well true. I moved it into the types.d.ts

File: src/components/editor/app-bar/help-button/help-button.tsx (hunk 1 above)
Comment (ErikMichelson) [reply src/components/editor/app-bar/help-button/help-button.tsx:9]: Ah okay.

**Review by ErikMichelson — APPROVED** [review]

**Review by DerMolly — APPROVED** [review]

**Comment by DerMolly** [comment]
Also it could be a good idea to extract all static links in a separate json file and import what we need from there. This way changing something along those lines is easiest in the future.

**Comment by ErikMichelson** [comment]
@DerMolly Wrong PR I guess...

**Pull request merged**